vms/platformvm/state/staker.go

```diff
@@ -121,3 +123,23 @@ func NewPendingStaker(txID ids.ID, staker txs.Staker) (*Staker, error) {
 		Priority:  staker.PendingPriority(),
 	}, nil
 }
+
+// While Staker object contains a staker's hot attributes, likely to be used pretty often
+// StakerColdAttributes contains a staker's cold attributes, rarely used, mostly when rewarding it.
+// Note that both Staker and StakerAttribute content comes from the stakerTx creating the staker.
+// In state.State we do have StakerMetadata information as well, which contains data about the stakers
+// that are generated during staker's activity (mostly uptimes)
+// TODO: consider moving delegatees reward here, out of StakersMetadata.
```

Comment:
this is the main idea here.
Ditching getTx is really a step towards classifying stakers attributes (all of them) into hotter and colder and arrange them in memory in a smart way